External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -5334,11 +5334,25 @@ void OpDispatchBuilder::ALUOpImpl(OpcodeArgs, FEXCore::IR::IROps ALUIROp, FEXCor
   else {
     Dest = LoadSource(GPRClass, Op, Op->Dest, Op->Flags, -1);
 
-    auto ALUOp = _Add(Dest, Src);
-    // Overwrite our IR's op type
-    ALUOp.first->Header.Op = ALUIROp;
+    /* On x86, the canonical way to zero a register is XOR with itself...
+     * because modern x86 detects this pattern in hardware. arm64 does not
+     * detect this pattern, we should do it like the x86 hardware would. On
+     * arm64, "mov x0, #0" is faster than "eor x0, x0, x0". Additionally this
+     * lets more constant folding kick in for flags.
+     */
+    if (ALUIROp == FEXCore::IR::IROps::OP_XOR &&
+        Op->Dest.IsGPR() && Op->Src[0].IsGPR() &&
+        memcmp(&Op->Dest.Data.GPR, &Op->Src[0].Data.GPR, sizeof(Op->Dest.Data.GPR)) == 0) {
```

Comment:
It's probably cleaner to default the <=> operator for the GPR struct here and then you can just do `Op->Dest.Data.GPR == Op->Src[0].Data.GPR`

Reply:
How do I do that? It's an anonymous struct

Reply:
You can just name the struct, don't think there's any particular reason to keep it unnamed

Reply:
Working on it

Reply:
There you go, with #2806 merged. you'll now just be able to do `Op->Dest.Data.GPR == Op->Src[0].Data.GPR` once rebased.